### PR TITLE
Enable nullable

### PR DESCRIPTION
**Pull request by iamcarbon**
# Types of Changes

This PR enables nullable annotations. No new APIs or functionality were introduced.

## Prerequisites

Please make sure you can check the following two boxes:

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project

## Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [x] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: src/AngleSharp/BrowsingContext.cs [hunk 1]
```diff
@@ -182,11 +182,11 @@ public IDocument Active
         /// <param name="name">The name of the child context, if any.</param>
         /// <param name="security">The security flags to apply.</param>
         /// <returns></returns>
-        public IBrowsingContext CreateChild(String name, Sandboxes security)
+        public IBrowsingContext CreateChild(String? name, Sandboxes security)
         {
             var context = new BrowsingContext(this, security);
 
-            if (!String.IsNullOrEmpty(name))
+            if (name is { Length: > 0 })
```
Comment (iamcarbon) [inline src/AngleSharp/BrowsingContext.cs:189]: Annoyingly, String.IsNullOrEmpty doesn't update the nullable status.

**Comment by iamcarbon** [comment]
Ready for review / feedback.

**Review by FlorianRappl — COMMENTED** [review]
This looks great! :beers:

I love it - only thing is right now the failing test. Not sure if that has to do with the PR or with the target. I'll look into that later.

**Comment by iamcarbon** [comment]
Quick update. This single test failure is caused by a connection issue to http://www.empireaerials.net. 

I attempted to build a custom handler to decompress a local gzip resource -- but realized we are testing the AutomaticDecompression behavior of the DefaultHttpRequester. Sadly, this utilizes the legacy WebRequest and WebResponse API -- which doesn't appear to have an easy method of mocking.

My recommendation is that we disable this test for now and consider swapping out the HttpWebRequest / HttpWebResponse  backed API with HttpClient in a new PR. With an HttpClient backed implementation, we can fully emulate the http pipeline within the testing framework using a custom HttpMessageHandler.

**Comment by FlorianRappl** [comment]
> My recommendation is that we disable this test for now and consider swapping out the HttpWebRequest / HttpWebResponse backed API with HttpClient in a new PR. With an HttpClient backed implementation, we can fully emulate the http pipeline within the testing framework using a custom HttpMessageHandler.

I think this would be rather something for the `HttpRequester` from `AngleSharp.Io`. For me its fine to disable it though.

**Review by FlorianRappl — APPROVED** [review]

**Pull request merged**